src/vmm/src/devices/virtio/net/device.rs

```diff
@@ -607,6 +607,17 @@ impl Net {
                     continue;
                 }
             };
+
+            // We only handle frames that are up to MAX_BUFFER_SIZE
+            if buffer.len() > MAX_BUFFER_SIZE {
+                error!("net: received too big frame from driver");
+                self.metrics.tx_malformed_frames.inc();
+                tx_queue
+                    .add_used(mem, head_index, 0)
+                    .map_err(DeviceError::QueueError)?;
```

Comment:
out of curiousity, why `QueueError`, not `MalformedPayload`?

Reply:
Because we don't fail, we just ignore that frame. The failure which we return is when we try to add the frame in the used queue and that fails

Reply:
ah, got it, thanks.